ADC based button source

markadr opened the pull request:
As discussed in #45, this change adds support for an analog device to be used as a button source. 

The new source type is defined as:
```
type: "adc",           // ADC based source
  adc: "<adc>",        // the adc entry, for example "pico_adc[0]", required
  threshold: <number>, // the adc threshold level to compare against the adc reading, default 0
  above: <bool>,       // true (default) -> read as ON when the adc reading is above the threshold
                       // false -> read as ON when the adc reading is below the threshold
```
An example configuration entry would be:
```
buttons: [
        {
            source: { type: "adc", adc: "pico_adc[1]", threshold: 3000, above: true },
            action: { type: "key", key: "up"}
        },
]
```

markadr commented:
@mjrgh let me know if you have any feedback.

mjrgh reviewed (commented):
Thanks!  This all looks pretty good.  One thing I'd change is the use of the native range for the ADC - I'd change that to use the normalized range instead.  I don't think there's any benefit for the user in using the device native range, since that's a detail of the device that most users would rather not need to know.  Using the normalized range would make that transparent to the user.

markadr commented:
@mjrgh thanks for the feedback, reworked the calls to make use of the nomalized value and adjusted the JSON reference to match the change. Also made the default `threshold` value 32767 from 0.

mjrgh commented:
Great!  I'll look at merging it shortly.

mjrgh commented:
Hi there.  I just merged this into my repos.  I did a little light reformatting to adjust it to my style conventions, and I also refactored the ADC config key enumeration to create a central routine (in the config manager) to do that work.  The new code needed for that in the button ADC source was essentially the same as the plunger code, so it's clearly something that should have been part of the ADC manager to begin with, but I just didn't realize that until this almost identical use case came up in this other context.

The merge is in my latest commit (https://github.com/mjrgh/PinscapePico/commit/93d26d3f4a89ff42d61ac54e1e8e305055d6d2e8).  If you have a chance to give it a try, it would be good to test it out to make sure I didn't break anything in your new code in my refactoring job.

Thanks again for the contribution - I'm sure this is something that other people will find useful, since there are lots of people who build hybrid pinball/video cabinets who probably have similar non-button controls they want to use.  It's always nice to have more options.  Apologies for taking so long to get to it.  I wanted to do that refactoring work, so I had to set aside a little time for this.
